lib/streamaggr/streamaggr.go

```diff
@@ -250,8 +256,12 @@ func newAggregatorsFromData(data []byte, pushFunc PushFunc, opts *Options) (*Agg
 	}
 
 	ms := metrics.NewSet()
+	if opts == nil {
```

Comment:
I think we should remove this check and pass &Options{} in test codes, since we should never pass `opts` as nil to `newAggregatorsFromData()`. Then if code panic, it exposes bug.

Reply:
I've moved this check out of `newAggregator` function, we're passing `opts = nil` in tests

Reply:
Yeah, that feels the same to me. I think this check now is only for UT, and if it's only for UT, maybe we don't need it, as it could masked possible bug in future code.

Reply:
@hagen1778 need your feedback here